src/components/common/YourCompany/ChangeBusinessType.vue

```diff
@@ -397,6 +392,64 @@ export default class ChangeBusinessType extends Mixins(CommonMixin) {
   protected submitTypeChange () {
     this.setEntityType(this.selectedEntityType)
     this.isEditingType = false
+
+    if (this.shouldUpdateName()) {
+      const originalName = this.getEntitySnapshot?.businessInfo.legalName
```

Comment:
397 -> 400 looks very similar to what we have in the store, might be a candidate for moving over to there.